CMake/FindSnappy.cmake

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
```

Comment:
are all these files only drop-in copies of the ones provided by the library? Or we make any modifications to them?

Reply:
I wrote them based off of the existing `Findglog.cmake` which was added [here](https://github.com/facebookincubator/velox/commit/2235149b7a3028f2175ebbcf44928d8c8a7890c3)